File: src/main/java/red/tetracube/iot/smartigloohub/data/entities/HouseAccount.java
package red.tetracube.iot.smartigloohub.data.entities;

import java.util.UUID;
import javax.persistence.*;

@Entity
@Table(schema = "igloo_hub", name = "houses_accounts")
public class HouseAccount {

    @Id
    private UUID id;

    @JoinColumn(name = "account_id", nullable = false)
    @ManyToOne(targetEntity = Account.class, fetch = FetchType.LAZY)
    private Account account;

    @JoinColumn(name = "house_id", nullable = false)
    @ManyToOne(targetEntity = House.class, fetch = FetchType.LAZY)
    private House house;

}

File: src/main/kotlin/red/tetracube/iot/smartigloohub/data/entities/HouseAccount.kt
package red.tetracube.iot.smartigloohub.data.entities

import java.util.UUID
import javax.persistence.*

@Entity
@Table(schema = "igloo_hub", name = "houses_accounts")
data class HouseAccount(
    @Id
    val id: UUID? = null,

    @JoinColumn(name = "account_id", nullable = false)
    @ManyToOne(targetEntity = Account::class, fetch = FetchType.LAZY)
    val account: Account? = null,

    @JoinColumn(name = "house_id", nullable = false)
    @ManyToOne(targetEntity = House::class, fetch = FetchType.LAZY)
    val house: House? = null
)
